FirebaseAppCheck/Sources/Core/FIRAppCheck.m

```diff
@@ -309,6 +321,39 @@ - (nonnull NSString *)notificationTokenKey {
   });
 }
 
+- (FBLPromise<FIRAppCheckToken *> *)retrieveLimitedTokenForcingRefresh {
+  return [FBLPromise do:^id _Nullable {
+    if (self.ongoingLimitedUseTokenPromise == nil) {
+      // Kick off a new operation only when there is not an ongoing one.
+      self.ongoingLimitedUseTokenPromise =
+          [self createRetrieveLimitedUseTokenPromiseForcingRefresh]
+
+              // Release the ongoing operation promise on completion.
+              .then(^FIRAppCheckToken *(FIRAppCheckToken *token) {
+                self.ongoingLimitedUseTokenPromise = nil;
+                return token;
+              })
+              .recover(^NSError *(NSError *error) {
+                self.ongoingLimitedUseTokenPromise = nil;
+                return error;
+              });
+    }
+    return self.ongoingLimitedUseTokenPromise;
+  }];
+}
+
+- (FBLPromise<FIRAppCheckToken *> *)createRetrieveLimitedUseTokenPromiseForcingRefresh {
+  return [self getLimitedUseTokenForcingRefresh].recover(^id _Nullable(NSError *_Nonnull error) {
+    return [self limitedUseToken];
+  });
+}
+
+- (FBLPromise<FIRAppCheckToken *> *)getLimitedUseTokenForcingRefresh {
+  FBLPromise *rejectedPromise = [FBLPromise pendingPromise];
+  [rejectedPromise reject:[FIRAppCheckErrorUtil cachedTokenNotFound]];
+  return rejectedPromise;
+}
```

Comment:
I believe you can remove these two methods -- it looks like `getLimitedUseTokenForcingRefresh` is just returning a rejected FBLPromise, so in `retrieveLimitedTokenForcingRefresh`, instead of invoking `createRetrieveLimitedUseTokenPromiseForcingRefresh`, you can just call `limitedUseToken` directly.

Reply:
yes, removed the two promises thx